### PR TITLE
feat: support compiling contract type JSONs

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Support compiling contract types.
I need this to better debug stuff in Ape regarding multichain and adding tests.

**had to make lots of doc edits to be able to create a good section for adding a note about the JSON compiler**

1. Noticed there was no compile guide so started one by extracting stuff out of the project guide and make a compile guide
2. As I was doing that, wanted to link to a different section regarding configuring plugins (regarding compiler plugins in this context) but there was not a guide for this either. And sort of misleadingly, the guide named plugins was only about creating plugins. Thus, I create a guide specifically for installing and configuring plugins, linked to it from the project guide, and then linked to it from the new compile guide I made
^ so that is why there are so many documentation changes

### How I did it

If it is not a list in the json compiler, assume it is a raw contract type

### How to verify it

* Compile a contract using vyper
* Grab the JSON of the contract type and move it to a new project
* Rename the source ID to be json-like and represent the new file
* Compile is using the JSON compiler

^ there is a test for this!

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: docs/userguides/developing_plugins.md [hunk 1]
```diff
@@ -1,9 +1,8 @@
 # Developing Plugins
 
-Your plugin project can be any type of python project, so long as its package name starts with `ape_` (such as 
-`ape_ethereum`). To create an `ape` plugin, implement one or more API classes from the the `ape.api` namespace or add 
-key `ape_cli_subcommands` to your entry-points list in your project's `setup.py`, depending on what type of plugin you 
-want to create. This guide is intended to assist in both of those use-cases.
+Your plugin project can be any type of python project, so long as its package name starts with `ape_` (such as `ape_ethereum`).
+To create an `ape` plugin, implement one or more API classes from the the `ape.api` namespace or add key `ape_cli_subcommands` to your entry-points list in your project's `setup.py`, depending on what type of plugin you want to create.
```
Comment (fubuloubu) [inline docs/userguides/developing_plugins.md:4]:
```suggestion
To create an `ape` plugin, implement one or more API classes from the the `ape.api` namespace and/or add key
 `ape_cli_subcommands` to your entry-points list in your project's `setup.py`, depending on what type of plugin you want to create.
```

File: docs/userguides/developing_plugins.md [hunk 2]
```diff
@@ -98,14 +94,11 @@ entry_points={
...
-group because it is logically separate from the Python module loading process. If you try to define them together and 
-use `ape` as a library as well, there is a race condition in the loading process that will prevent the CLI plugin from 
-working.
+**NOTE**: Typically, a `_cli.py` module is used instead of a `__init__.py` module for the location of the Click CLI group because it is logically separate from the Python module loading process.
+If you try to define them together and use `ape` as a library as well, there is a race condition in the loading process that will prevent the CLI plugin from working.
 
-For common `ape-click` usages, use the `ape.cli` namespace. For example, use the 
-[@existing_alias_argument() decorator](../methoddocs/cli.html#ape.cli.arguments.existing_alias_argument)) when you need 
-a CLI argument for specifying an existing account alias:
+For common `ape-click` usages, use the `ape.cli` namespace.
```
Comment (fubuloubu) [inline docs/userguides/developing_plugins.md:100]: Freudian slip?
```suggestion
For common `click` usages, use the `ape.cli` namespace.
```

File: docs/userguides/installing_plugins.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Ape ships with core plugins to help Ape work out-of-the-box.
+To see the core plugins that come with Ape, run the following command:
+
+```bash
+ape plugins list --all
+```
+
+Normally, the `ape plugins list` command shows you all the plugins you have installed.
+However, when you include the `--all` flag, it shows the core plugins and the available plugins as well.
+**NOTE**: The available plugins list is trusted and from the Apeworx organziation, however you can install third-party plugins as well.
```
Comment (fubuloubu) [inline docs/userguides/installing_plugins.md:22]:
```suggestion
**NOTE**: The available plugins list is trusted and from the ApeWorX organization, however you can install third-party plugins from other sources as well.
```

**Review by fubuloubu — APPROVED** [review]

File: docs/userguides/developing_plugins.md [hunk 4]
```diff
@@ -134,11 +127,11 @@ my_trezor_account = accounts.load("trezor_0")  # Created using the 'ape-trezor'
...
 
 ```bash
-ape run my_script --network ethereum:local:my_provider_plugin
+ape console my_script --network ethereum:local:my_provider_plugin
 ```
 
-**NOTE**: The `--network` option is available on the commands `run`, `test`, and `console` or any CLI command that 
-uses the [network option decorator](../methoddocs/cli.html?highlight=network_option#ape.cli.options.network_option).
+**NOTE**: The `--network` option is available on the commands `test` and `console` as well as any CLI command that uses the [network option decorator](../methoddocs/cli.html?highlight=network_option#ape.cli.options.network_option).
+To learn more about networking in Ape, follow [this guide](./networks.html).
```
Comment (NotPeopling2day) [inline docs/userguides/developing_plugins.md:134]:
```suggestion
To learn more about networks in Ape, follow [this guide](./networks.html).
```

File: docs/userguides/projects.md [hunk 5]
```diff
@@ -1,48 +1,50 @@
...
 
+Notice that you can configure you ape project using the `ape-config.yaml` file.
+See the [configuration guide](./config.html) for a more detailed explanation of settings you can adjust.
 
- [Configuration guide](../userguides/config.md) for a more detailed explanation of settings you can
-use in your `ape-config.yaml` files.
+## Adding Plugins
+
+Your project may require plugins.
+To install plugins, use the `ape plugins install` command.
```
Comment (NotPeopling2day) [inline docs/userguides/projects.md:23]: Should we add the `.` with the command here?
```suggestion
To install plugins, use the `ape plugins install .` command.
```

**Review by NotPeopling2day — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**